languages/python/sqlalchemy-oso/sqlalchemy_oso/auth.py

```diff
@@ -38,24 +38,46 @@ def authorize_model(oso: Oso, actor, action, session: Session, model) -> Query:
     :param session: The SQLAlchemy session.
     :param model: The model to authorize, must be a SQLAlchemy model.
     """
-    # TODO (dhatch): More robust name mapping?
+    filters = authorize_model_filter(oso, actor, action, session, model)
+    if filters is None:
+        return session.query(model)
+
+    return session.query(model).filter(filters)
+
+
+def authorize_model_filter(oso: Oso, actor, action, session: Session, model):
+    """Return SQLAlchemy expression that applies the policy to ``model``.
+
+    Executing this query will return only authorized objects. If the request is
+    not authorized, a query that always contains no result will be returned.
+
+    :param oso: The oso class to use for evaluating the policy.
+    :param actor: The actor to authorize.
+    :param action: The action to authorize.
 
+    :param session: The SQLAlchemy session.
+    :param model: The model to authorize, must be a SQLAlchemy model.
+    """
+    # TODO (dhatch): Check that model is a model.
+    # TODO (dhatch): More robust name mapping?
```

Comment:
You removed this comment from `authorize_model` -- does it still apply here?

Reply:
Yes, authorize model calls this, so we only need it in one place.